File: challenges/aoc/2023/1/1.cpp
#include <ctype.h>

#include <chrono>
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

int main() {
    // ifstream file("sample.txt");
    ifstream file("input.txt");

    // for testing performance
    chrono::steady_clock::time_point start = chrono::steady_clock::now();

    string line;
    int count = 0;

    while (getline(file, line)) {
        char first = 'x';
        char last = 'x';

        for (int i = 0; i < line.length(); i++) {
            char currentChar = line[i];
            if (isdigit(currentChar)) {
                if (first == 'x') {
                    first = currentChar;
                } else
                    last = currentChar;
            }
        }

        if (last == 'x') {
            last = first;
        }

        int n = ((first - '0') * 10) + (last - '0');
        count += n;
    }

    file.close();

    cout << "result: " << count;

    // for testing performance
    chrono::steady_clock::time_point end = chrono::steady_clock::now();
    chrono::duration<double> elapsedSeconds = end - start;
    cout << "Execution Time: " << elapsedSeconds.count() << " seconds" << endl;

    return 0;
}

File: challenges/aoc/2023/1/1.py
import time

with open("input.txt", "r") as file:
    lines = file.readlines()

# only for testing performance
start_time = time.time()

count = 0

for line in lines:
    first = 'x'
    last = 'x'

    for char in line:
        if char.isdigit():
            if first == 'x':
                first = char
            else:
                last = char

    if last == 'x':
        last = first

    n = first+last
    count += int(n)

print(f"result: {count}")

# only for testing performance
end_time = time.time()
execution_time = end_time - start_time
print("Execution Time:", execution_time, "seconds")
